clang-tools-extra/clangd/refactor/tweaks/DefineOutline.cpp

```diff
@@ -407,10 +431,21 @@ class DefineOutline : public Tweak {
       return !SameFile;
     }
 
-    // Bail out in templated classes, as it is hard to spell the class name,
-    // i.e if the template parameter is unnamed.
-    if (MD->getParent()->isTemplated())
-      return false;
+    for (const CXXRecordDecl *Parent = MD->getParent(); Parent;
+         Parent =
+             llvm::dyn_cast_or_null<const CXXRecordDecl>(Parent->getParent())) {
+      if (auto Params = Parent->getDescribedTemplateParams()) {
```

Comment:
Please explicitly write the type, it is not clear from the rhs what it is (`const TemplateParameterList *`)

Reply:
Done, but note that auto is not generally used in this project the way you prefer (with the type obvious from the rhs).

Reply:
That's from the LLVM Coding Standards, which are not a hard ruleset of course. In Clang-Tidy we do follow the rule regarding `auto` so I just suggested it here as well.